app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_user_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
- [ ] [optional] @SaneMyburg Great work adding a method to count the number of posts, it would be even better if you used [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_user_posts_counter` method every time a new post has been saved to the database, then make the `update_user_posts_counter` method private.
```suggestion
 after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end

```